contracts/exchange-forwarder/contracts/src/MixinAssets.sol

```diff
@@ -50,6 +50,26 @@ contract MixinAssets is
         _transferAssetToSender(assetData, amount);
     }
 
+    /// @dev Approves the respective proxy for a given asset to transfer tokens on the Forwarder contract's behalf.
+    ///      This is necessary because an order fee denominated in the maker asset (i.e. a percentage fee) is sent by the
+    ///      Forwarder contract to the fee recipient.
+    ///      This method needs to be called before forwarding orders of a maker asset that hasn't
+    ///      previously been approved.
+    /// @param assetData Byte array encoded for the respective asset proxy.
+    function approveMakerAssetProxy(bytes calldata assetData)
```

Comment:
I wonder if it would make sense to have another function that sets a user-specified approval value.